### PR TITLE
Compile with gcc 9

### LINKED ISSUE #2415: Support C++20
### Description of the task
C++20 features are cool and we want to use them. More info: https://en.cppreference.com/w/cpp/20. 

### Acceptance criteria

- [ ] upgrade gcc or clang to support C++20

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* Cleaned up remaining ARM
* Upgrade from GCC-7 to GCC-9
* Replace experimental/filesystem with filesystem
* Upgrade Jetson nano gcc-7 to gcc-9

### Resolved Issues
resolves #2415

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/handheld_control_main.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ int main(int argc, char **argv)
     auto backend = std::make_shared<WifiBackend>(thunderbots_config->getBackendConfig());
 
     controller->Subject<ControllerInput>::registerObserver(primitive_generator);
-    primitive_generator->Subject<TbotsProto::PrimitiveSet>::registerObserver(backend);
+    // primitive_generator->Subject<TbotsProto::PrimitiveSet>::registerObserver(backend);
```
Comment (itsarune) [inline src/software/handheld_control_main.cpp]: why commented out?

**Review by itsarune — COMMENTED** [review]
nice to have this in, one tiny thing

File: src/software/handheld_control_main.cpp (hunk 1 above)
Comment (jonathanlew) [reply src/software/handheld_control_main.cpp]: nice catch, fixed

**Review by itsarune — DISMISSED** [review]

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
`./thunderloop_main: /lib/aarch64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by ./thunderloop_main)`

doesn't seem to work out of the box on the jetson nano unfortunately :(

**Review by akhilveeraghanta — APPROVED** [review]

**Review by itsarune — APPROVED** [review]

**Pull request merged**